### PR TITLE
GET | /users

### PR DESCRIPTION
**Pull request by Alaalser**
Relates #26
- Create getUsers controller function
- Made a query to select users data
- Create the get users route
- Create the test for getUsers route

### CODE REVIEW

File: server/database/queries/users/getUsers/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const connection = require('../../../data/connection');
+
+const getAllUsers = () => {
+  const sql = {
+    text: 'SELECT * FROM users',
```
Comment (Mu7ammadAbed) [inline server/database/queries/users/getUsers/index.js:5]: This query returns everything in the user's columns even the password and that's not a thing we want to return a response with

File: server/controllers/users/getUsers/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+const getUsers = async (req, res, next) => {
+  try {
+    const { rows: data } = await getAllUsers();
+    res.status(200).json({ statusCode: 200, data });
+  } catch (err) {
+    next(err);
+  }
+};
+
+module.exports = { getUsers };
```
Comment (Mu7ammadAbed) [inline server/controllers/users/getUsers/index.js:12]:
```suggestion
module.exports = getUsers;
```

File: server/controllers/users/index.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const { getUsers } = require('./getUsers');
```
Comment (Mu7ammadAbed) [inline server/controllers/users/index.js:1]:
```suggestion
const getUsers = require('./getUsers');
```

File: server/controllers/users/getUsers/index.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const { getAllUsers } = require('../../../database/queries');
```
Comment (Mu7ammadAbed) [inline server/controllers/users/getUsers/index.js:1]: Rename this module to `getUsersQuery` or rename this module name to `getAllUsers`

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: .eslintrc.json [hunk 5]
```diff
@@ -15,6 +15,7 @@
       {
         "singleQuote": true
       }
-    ]
+    ],
+    "jest/no-done-callback": "off"
   }
```
Comment (Osama-you) [inline .eslintrc.json:20]: remove that rule is not needed anymore

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Osama-you — APPROVED** [review]

**Review by Israa91 — APPROVED** [review]

**Review by alaa-sahloub — APPROVED** [review]

**Pull request merged**